src/precompile.jl

```diff
@@ -0,0 +1,14 @@
+"""
+    _precompile_()
+
+Run some tiny workloads that the package should precompile.
+"""
+function _precompile_()
```

Comment:
As is currently written, the `HTTP.get` call won't actually result in any code being cached, since it's just being run as-is (and will thus incur the same cost every time the package is loaded). You instead want to include a "generating precompilation code" directive to enable code being cached, like:
```julia
function _precompile_()
    ccall(:jl_generating_output, Cint, ()) == 1 || return nothing
    # ...
```

Reply:
Actually, that `ccall` line is just to prevent it from running if you launch julia with `--compiled-modules=no`. Basically what happens is:
- [during precompilation] Julia parses & executes the package code. This defined the module
- [during precompilation] Julia saves a "snapshot" of the module in the `*,ji` file stored in your `.julia/compile/v1.x`. This includes Methods and any top-level variables.
- [during `using HTTP`] Julia restores the snapshot

Consequently, "work" that gets done at toplevel when the source files are executed doesn't get re-run when you load from the cache file. But with `--compiled-modules=no`, Julia will avoid loading your `*.ji` files.